### PR TITLE
TODO - Updated NavigationState to common types with Hashable

### PR DESCRIPTION
**Pull request by Archdoog**

### CODE REVIEW

File: common/ferrostar/src/models.rs [hunk 1]
```diff
@@ -59,18 +59,51 @@ impl From<Rect> for BoundingBox {
...
 }
 
+/// The heading of the user/device.
+/// 
+/// Ferrostar prefers course over ground
+#[derive(Clone, Copy, PartialEq, PartialOrd, Debug, uniffi::Record)]
+pub struct Heading {
+    pub geographic_heading: f64,
+    pub accuracy: f64,
+    pub x: f64,
```
Comment (Archdoog) [inline common/ferrostar/src/models.rs]: Convert this to an optional `RawHeading` type.

File: common/ferrostar/src/models.rs [hunk 2]
```diff
@@ -59,18 +59,51 @@ impl From<Rect> for BoundingBox {
...
 /// The direction in which the user/device is observed to be traveling.
 #[derive(Clone, Copy, PartialEq, PartialOrd, Debug, uniffi::Record)]
 pub struct CourseOverGround {
     /// The direction in which the user's device is traveling, measured in clockwise degrees from
     /// true north (N = 0, E = 90, S = 180, W = 270).
-    pub degrees: u16,
+    pub degrees: f64,
     /// The accuracy of the course value, measured in degrees.
-    pub accuracy: u16,
+    pub accuracy: f64,
```
Comment (Archdoog) [inline common/ferrostar/src/models.rs]: Using an integer reduces edge cases/invalid/etc.

File: common/ferrostar/src/models.rs [hunk 3]
```diff
@@ -59,18 +59,51 @@ impl From<Rect> for BoundingBox {
...
+    pub x: f64,
+    pub y: f64,
+    pub z: f64,
+    pub timestamp: SystemTime,
+}
+
+impl Heading {
+    pub fn new(
+        geographic_heading: f64,
```
Comment (Archdoog) [inline common/ferrostar/src/models.rs]: Convert to `true_heading`. Convert to int and add docstring.

Comment (Archdoog) [reply common/ferrostar/src/models.rs]: And switching to unsigned int avoids the negative invalid heading value.

File: common/ferrostar/src/models.rs (hunk 1 above)
Comment (Archdoog) [reply common/ferrostar/src/models.rs]: Drop this for now, we can add it later.

File: common/ferrostar/src/models.rs (hunk 3 above)
Comment (Archdoog) [reply common/ferrostar/src/models.rs]: Add one or more unit tests to test a negative here to see if it gracefully fails when iOS gives us -1 "invalid"
